third_party/maya/lib/usdMaya/writeJob.cpp

```diff
@@ -350,6 +350,65 @@ UsdMaya_WriteJob::_BeginWriting(const std::string& fileName, bool append)
         }
     }
 
+    // If stripNamespaces is on, this validates that we don't have a
```

Comment:
For readability we'd prefer this check to be in a separate static function above instead of in a large lambda here. Perhaps something like:
```
namespace {

// If stripNamespaces is on, this validates that we don't have a
// name collision as a result (and maintains mUsdPathToDagPathMap,
// which is only used to do this validity / name collision check,
// and so is only populated if stripNamespaces is on)
static
bool
_CheckStrippedNamespaceIsUnique(
    const UsdMayaPrimWriter& primWriter,
    const UsdPrim& usdPrim,
    const UsdMayaJobExportArgs& jobArgs,
    const UsdMayaWriteJobContext& jobContext,
    TfHashMap<SdfPath, MDagPath, SdfPath::Hash>* usdPathToDagPathMap)
{
    // ...
}

} // end anonymous namespace
```